From 757fb74c4b51f31b11bdbc5f2ae948a08dbc9c46 Mon Sep 17 00:00:00 2001
From: Venkat Devarasetty <vdevaras@codeaurora.org>
Date: Wed, 5 Nov 2014 14:29:56 +0530
Subject: [PATCH] msm: hotplug: Do not skip the platform init for boot core

First warmboot for secondary cores is treated as cold boot
and platform secondary init functionality is skipped. But
for boot core the first warmboot is real warmboot so do the
platform init  even for frist time for this core. This change
is necessary for core0 hotplug support.

Change-Id: I736e01f3ac3d7fa54eb2d70ad1f813acbebb3d16
Signed-off-by: Venkat Devarasetty <vdevaras@codeaurora.org>
---
 arch/arm/mach-msm/hotplug.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-msm/hotplug.c b/arch/arm/mach-msm/hotplug.c
index ddd5e9c..d44094f 100644
--- a/arch/arm/mach-msm/hotplug.c
+++ b/arch/arm/mach-msm/hotplug.c
@@ -142,7 +142,8 @@ int msm_platform_secondary_init(unsigned int cpu)
 
 	if (!(*warm_boot)) {
 		*warm_boot = 1;
-		return 0;
+		if (cpu)
+			return 0;
 	}
 	msm_jtag_restore_state();
 #if defined(CONFIG_VFP) && defined (CONFIG_CPU_PM)
